runtime/src/main/java/com/redhat/ipaas/rest/SortOptionsFromQueryParams.java

```diff
@@ -0,0 +1,54 @@
+/*
+ *
+ * Copyright 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.redhat.ipaas.rest;
+
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.UriInfo;
+
+import com.redhat.ipaas.rest.util.SortOptions;
+
+/**
+ * @author roland
+ * @since 14/12/16
+ */
+class SortOptionsFromQueryParams implements SortOptions {
+    private final MultivaluedMap<String, String> queryParams;
+
+    private static final String SORT_FIELD = "sortField";
+    private static final String SORT_DIRECTION = "sortDir";
+
+
+    SortOptionsFromQueryParams(UriInfo uri) {
+        queryParams = uri.getQueryParameters();
```

Comment:
Just extract the sort field & dir in the constructor. No need for calculating in each getter.